Legacy/bonej/src/main/java/org/bonej/plugins/Orienteer.java

```diff
@@ -224,38 +226,263 @@ public Orienteer() {
+		super.windowActivated(e);
+		update();
+		WindowManager.setWindow(this);
+	}
+
+	@Override
+	public void windowClosing(final WindowEvent e) {
+		close();
+		Prefs.saveLocation(LOC_KEY, getLocation());
+	}
+
+	private void addLabels() {
+		if (WindowManager.getImageCount() == 0) return;
+		final ImagePlus imp = WindowManager.getImage(activeImpID);
+		scale = imp.getCanvas().getMagnification();
+		final Font font = new Font("SansSerif", Font.PLAIN, (int) (fontSize /
+			scale));
+		final double lsinTheta = (length + (fontSize / scale)) * Math.sin(theta);
+		final double lcosTheta = (length + (fontSize / scale)) * Math.cos(theta);
+		addString(directions[0], (int) (p.x + lsinTheta), (int) (p.y - lcosTheta),
+			Color.RED, font);
+		addString(directions[1], (int) (p.x - lsinTheta), (int) (p.y + lcosTheta),
+			Color.BLUE, font);
+		addString(directions[2], (int) (p.x + lcosTheta), (int) (p.y + lsinTheta),
+			Color.BLUE, font);
+		addString(directions[3], (int) (p.x - lcosTheta), (int) (p.y - lsinTheta),
+			Color.BLUE, font);
+	}
+
+	private void addPath(final Shape shape, final BasicStroke stroke) {
+		final Roi roi = new ShapeRoi(shape);
+		roi.setStrokeColor(Color.BLUE);
+		roi.setStroke(stroke);
+		roi.setStrokeWidth(roi.getStrokeWidth() / (float) scale);
+		overlay.add(roi);
+	}
+
+	private void addString(final String text, final int x, final int y,
+		final Color color, final Font font)
+	{
+		final TextRoi roi = new TextRoi(x, y, text, font);
+		roi.setLocation(x - text.length() * (int) (fontSize / scale) / 4, y -
+			(int) (fontSize / scale) / 2);
+		roi.setStrokeColor(color);
+		overlay.add(roi);
+	}
+
+	/**
+	 * Keep hash lists up to date by removing keys of closed windows
+	 */
+	private void checkHash() {
+		final Set<Integer> idset = thetaHash.keySet();
+		for (final Integer i : idset) {
+			if (WindowManager.getImage(i) == null) clearHashes(i);
+		}
+	}
+
+	/**
+	 * Check if an image is already handled by Orientation
+	 *
+	 * @param imp an image.
+	 * @return true if this image is already handled by Orientation
+	 */
+	private boolean checkHash(final ImagePlus imp) {
+		final Integer i = imp.getID();
+		return thetaHash.containsKey(i);
+	}
+
+	/**
+	 * Remove all image IDs from the hashes.
+	 */
+	private void clearHashes() {
+		thetaHash.clear();
+		reflectHash.clear();
+		pathHash.clear();
+		unitHash.clear();
+		axisHash.clear();
+		centreHash.clear();
+		lengthHash.clear();
+	}
+
+	/**
+	 * Remove a single image ID from the hashes.
+	 */
+	private void clearHashes(final Integer i) {
+		thetaHash.remove(i);
+		reflectHash.remove(i);
+		pathHash.remove(i);
+		unitHash.remove(i);
+		axisHash.remove(i);
+		centreHash.remove(i);
+		lengthHash.remove(i);
+	}
+
+	/**
+	 * Get the orientation of the principal direction
+	 *
+	 * @param imp an image.
+	 * @return Orientation in radians clockwise from 12 o'clock
+	 */
+	private double getOrientation(final ImagePlus imp) {
+		final Integer id = imp.getID();
```

Comment:
3 implementations of `getOrientation` do pretty different things depending on signature